From d11e71a104a2dd27b108520ff33d3f6792c7986a Mon Sep 17 00:00:00 2001
From: Garrett Brown <garrett.brown@aclima.io>
Date: Tue, 17 Mar 2020 22:28:20 -0700
Subject: [PATCH] CMake: Provide missing android_log module

The android_log CMake module is provided by the Hunter build system, so we
need to import this file for building without Hunter.
---
 CMakeLists.txt                      |  3 +++
 cmake/modules/Findandroid_log.cmake | 32 +++++++++++++++++++++++++++++
 2 files changed, 35 insertions(+)
 create mode 100644 cmake/modules/Findandroid_log.cmake

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c500aec..3449203 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,9 @@ set(CMAKE_TOOLCHAIN_FILE
 include("cmake/Hunter/init.cmake")
 project(libp2p C CXX)
 
+# Expose cmake modules for finding packages
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 option(TESTING "Build tests" ON)
diff --git a/cmake/modules/Findandroid_log.cmake b/cmake/modules/Findandroid_log.cmake
new file mode 100644
index 0000000..67158b3
--- /dev/null
+++ b/cmake/modules/Findandroid_log.cmake
@@ -0,0 +1,32 @@
+#.rst:
+# Findandroid_log.cmake
+# ----------
+# Finds the Android log library. Based on a helper function in the Hunter
+# package manager.
+#
+# This will define the following imported target:
+#
+#   android_log::android_log - The Android log library
+#
+
+set(ANDROID_LOG_TARGET_NAME "android_log::android_log")
+
+find_library(ANDROID_LOG_LIBRARY "log")
+find_path(ANDROID_LOG_HEADER_DIR "android/log.h")
+
+if (NOT TARGET ${ANDROID_LOG_TARGET_NAME})
+  message(STATUS "Creating target '${ANDROID_LOG_TARGET_NAME}':")
+  message(STATUS "* library: '${ANDROID_LOG_LIBRARY}'")
+  message(STATUS "* header dir: '${ANDROID_LOG_HEADER_DIR}'")
+
+  add_library(${ANDROID_LOG_TARGET_NAME} UNKNOWN IMPORTED)
+  set_target_properties(
+    ${ANDROID_LOG_TARGET_NAME}
+    PROPERTIES
+    IMPORTED_LINK_INTERFACE_LANGUAGE "CXX"
+    IMPORTED_LOCATION "${ANDROID_LOG_LIBRARY}"
+    INTERFACE_INCLUDE_DIRECTORIES "${ANDROID_LOG_HEADER_DIR}"
+  )
+endif()
+
+mark_as_advanced(ANDROID_LOG_LIBRARY ANDROID_LOG_HEADER_DIR)
-- 
2.20.1

